packages/subgraph/src/index.ts

```diff
@@ -193,23 +196,46 @@ ponder.on('Space:SwapExecuted', async ({ event, context }) => {
     }
 
     try {
-        const existing = await context.db.sql.query.swap.findFirst({
-            where: eq(schema.swap.txHash, transactionHash),
+        // Calculate ETH amount for analytics
+        let ethAmount = 0n
+        if ((event.args.tokenIn as string).toLowerCase() === ETH_ADDRESS) {
+            ethAmount = event.args.amountIn
+        } else if ((event.args.tokenOut as string).toLowerCase() === ETH_ADDRESS) {
+            ethAmount = event.args.amountOut
+        }
+
+        // Write to swap table
+        await context.db.insert(schema.swap).values({
+            txHash: transactionHash,
+            spaceId: spaceId,
+            recipient: event.args.recipient,
+            tokenIn: event.args.tokenIn,
+            tokenOut: event.args.tokenOut,
+            amountIn: event.args.amountIn,
+            amountOut: event.args.amountOut,
+            poster: event.args.poster,
+            blockTimestamp: blockTimestamp,
+            createdAt: blockNumber,
         })
-        if (!existing) {
-            await context.db.insert(schema.swap).values({
-                txHash: transactionHash,
-                spaceId: spaceId,
-                recipient: event.args.recipient,
+
+        await context.db.insert(schema.analyticsEvent).values({
```

Comment:
Though this is in effect an "append-only" table and design, we are dealing with postgres and if a pk constraint fires here, it will possibly crash the indexer process so we should perform some conflict resolution before inserting.